### PR TITLE
feat: Make comment RegExps non-greedy to prevent some max call stack errors

### PR DESCRIPTION
**Pull request by Delagen**

### CODE REVIEW

**Comment by thlorenz** [comment]
This looks interesting and I'm fine with this in general as long as it doesn't break any other functionality.
It's been a while since you filed this (I'm sorry for the super long delay).
Is this still a problem for you or did the newer v8 version fix it?

**Comment by chpio** [comment]
> Is this still a problem for you or did the newer v8 version fix it?

just got the

> RangeError: Maximum call stack size exceeded

exception. NodeJs@14.18.3

File: .travis.yml [hunk 1]
```diff
@@ -4,4 +4,7 @@ node_js:
   - "0.12"
   - "4"
   - "6"
+  - "8"
+  - "10"
+  - "12"
```
Comment (phated) [inline .travis.yml:9]:
```suggestion
```

Going to remove these because we should be removing travis with #77

**Comment by thlorenz** [comment]
Thanks @phated for cleaning this up. If no tests are failing we could merge.
Just have to be very careful whenever we change the regex as it could fail in unexpected ways.

Therefore if we do merge we need to publish as new major (I'll take care of that).

**Comment by phated** [comment]
> Therefore if we do merge we need to publish as new major (I'll take care of that).

I agree! I was hoping to land #74 as a minor and then a few of the other open PRs as a major. Does that work for you @thlorenz ?

**Comment by thlorenz** [comment]
Yes that would work .. also now we got CI going again.

However the Node.js versions tested are super old now, we should probably test 0.14, 0.16, 0.18 instead.
So I'd like to get that in first, then merge #74, publish a minor and then worry about the regex PRs

**Comment by phated** [comment]
> However the Node.js versions tested are super old now, we should probably test 0.14, 0.16, 0.18 instead.

We're actually testing pretty much every major version of node now: `"0.10", "0.12", "4", "6", "8", "10", "12", "14", "16", "18"`

**Comment by thlorenz** [comment]
Ah ok didn't see CI running those versions.
So not sure at this point cause this PR is at odds with #75.

It seems like they should be combined into one (possibly making the regex recommended in #75 non-greedy)?

Maybe @Delagen has an idea as well?

**Comment by phated** [comment]
Yeah, I am planning to merge this into master and then update #75 to take it into account.

**Comment by thlorenz** [comment]
LGTM to merge, so go ahead once we publish another minor with the old regex.
Should I go ahead and publish a minor at this point?

**Comment by phated** [comment]
> Should I go ahead and publish a minor at this point?

Yep! I think the final minor is ready now.

**Comment by thlorenz** [comment]
OK, published 1.9 + merged CI PR. Now we can focos on that regex stuff :)

**Pull request merged**